### PR TITLE
add deprecation notice for mapbox traces

### LINKED ISSUE #4730: Add deprecation warning for Mapbox traces
Following #4726, add a deprecation warning to the Python console for Mapbox traces. Something like:
> `px.scatter_mapbox` is deprecated and will be removed in a future release. Please use `px.scatter_map` with MapLibre instead. For more information on migrating to MapLibre, refer to the migration guide [LINK].

### PR DESCRIPTION
**Pull request by archmoj**
Closes #4730.

### CODE REVIEW

**Review by gvwilson — APPROVED** [review]

**Comment by LiamConnors** [comment]
These show up when you are using the new traces

image

**Comment by gvwilson** [comment]
cc @archmoj

File: packages/python/plotly/plotly/graph_objs/_choroplethmapbox.py [hunk 1]
```diff
@@ -3,6 +3,11 @@
 
 
 class Choroplethmapbox(_BaseTraceType):
+    print(
+        "*choroplethmapbox* trace is deprecated!",
+        "Please consider switching to the *choroplethmap* trace type and `map` subplots.",
+        "Learn more at: https://plotly.com/javascript/maplibre-migration/",
```
Comment (ndrezn) [inline packages/python/plotly/plotly/graph_objs/_choroplethmapbox.py:9]: All links should point to the Python documentation: https://plotly.com/python/mapbox-to-maplibre/

**Comment by LiamConnors** [comment]
could we use a warning as mentioned by @ndrezn  above https://github.com/plotly/plotly.py/pull/4783#discussion_r1791936053

**Comment by archmoj** [comment]
> could we use a warning as mentioned by @ndrezn above [#4783 (comment)](https://github.com/plotly/plotly.py/pull/4783#discussion_r1791936053)

I am not sure is that's a good idea. 
My concern is that it could be too annoying for users and it may be considered a notable undesirable change.

**Comment by emilykl** [comment]
> > could we use a warning as mentioned by @ndrezn above [#4783 (comment)](https://github.com/plotly/plotly.py/pull/4783#discussion_r1791936053)
> 
> I am not sure is that's a good idea. My concern is that it could be too annoying for users and it may be considered a notable undesirable change.

@archmoj Being annoying is the whole point, so that users will change their code before the traces are removed entirely.

**Comment by LiamConnors** [comment]
@archmoj, I see the deprecation notice on traces and px functions that don't use mapbox

```
import plotly.express as px
df = px.data.gapminder().query("year == 2007")
fig = px.line_geo(df, locations="iso_alpha",
                  color="continent", # "continent" is one of the columns of gapminder
                  projection="orthographic")
fig.show()
```
image

image

But not on ones that do:

image

**Comment by emilykl** [comment]
@archmoj Maybe you're already on it but I think the warning needs to go inside the `__init__` function rather than at the top of the class.

**Comment by archmoj** [comment]
> @archmoj Maybe you're already on it but I think the warning needs to go inside the `__init__` function rather than at the top of the class to fix the issue Liam is seeing.

@LiamConnors Thanks for catching that. Let's try @emilykl's suggestion.

**Comment by LiamConnors** [comment]
@archmoj, I still see the issue

image

**Comment by archmoj** [comment]
@LiamConnors Debugging the bug you noticed wasn't easy. I figured the error is coming from the plotly.py templates which include defaults for "scattermapbox" traces as well as `"mapbox" subplots. 
On the other hand resolving the merge conflicts of this PR is not easy. 
So I close this PR; then open a new PR.
Thank you!

**Pull request closed without merging**